pkg/controllers/federate/controller.go

```diff
@@ -93,117 +89,147 @@ func (c *FederateController) IsControllerReady() bool {
 }
 
 func NewFederateController(
-	typeConfig *fedcorev1a1.FederatedTypeConfig,
 	kubeClient kubeclient.Interface,
 	dynamicClient dynamicclient.Interface,
-	federatedObjectInformer informers.GenericInformer,
-	sourceObjectInformer informers.GenericInformer,
+	fedClient fedclient.Interface,
+	fedObjectInformer fedcorev1a1informers.FederatedObjectInformer,
+	clusterFedObjectInformer fedcorev1a1informers.ClusterFederatedObjectInformer,
+	informerManager informermanager.InformerManager,
 	metrics stats.Metrics,
 	workerCount int,
 	fedSystemNamespace string,
```

Comment:
It seems like `fedSystemNamespace` is unused, maybe we can remove it?

Reply:
I meant to use it to filter out the `kube-admiral-system` objects which we should not federate. Updated the code to include that logic.